gazebo_ros/src/gazebo_ros_init.cpp

```diff
@@ -66,20 +66,12 @@ void GazeboRosInit::Load(int argc, char ** argv)
     "/clock",
     rclcpp::QoS(rclcpp::KeepLast(10)).transient_local());
 
-  // Get publish rate from parameter if set
-  rclcpp::Parameter rate_param;
-  if (impl_->ros_node_->get_parameter("publish_rate", rate_param)) {
-    if (rclcpp::ParameterType::PARAMETER_DOUBLE == rate_param.get_type()) {
-      impl_->throttler_ = Throttler(rate_param.as_double());
-    } else if (rclcpp::ParameterType::PARAMETER_INTEGER == rate_param.get_type()) {
-      impl_->throttler_ = Throttler(rate_param.as_int());
-    } else {
-      RCLCPP_WARN(impl_->ros_node_->get_logger(),
-        "Could not read value of param publish_rate [%s] as double/int, using default %ghz.",
-        rate_param.value_to_string().c_str(),
-        GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY);
-    }
-  }
+  // Publish rate parameter
+  impl_->ros_node_->declare_parameter("publish_rate",
+    rclcpp::ParameterValue(GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY));
+
+  auto rate_param = impl_->ros_node_->get_parameter("publish_rate");
```

Comment:
`declare_parameter` returns the value that after considering the default and the overrides, so this could be reduced to:

```c++
auto rate_param = impl_->ros_node_->declare_parameter(
  "publish_rate",
  rclcpp::ParameterValue(GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY));
```

And I may be wrong (not 100% sure of the type of `GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY`), but it should be possible to just do this:

```c++
double rate_param = impl_->ros_node_->declare_parameter(
  "publish_rate",
  GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY);
```

Or if it is not a double:

```c++
double rate_param = impl_->ros_node_->declare_parameter<double>(
  "publish_rate",
  GazeboRosInitPrivate::DEFAULT_PUBLISH_FREQUENCY);
```

Using this signature:

https://github.com/ros2/rclcpp/blob/8ff51833ad7bceeb4c4cb2bacfbe0de02da25e5e/rclcpp/include/rclcpp/node.hpp#L422-L428